File: proposed-projects/round-trip-experiment/reverse/20260909T040730Z_0947f229/20260910T065554Z_5d586b52/ported_implementation/src/utils/errors/grammar-parse-error.ts
/** Port of `gbnf/utils/errors/grammar_parse_error.py`. */

import { buildErrorPosition } from './build-error-position.js';

export const GRAMMAR_PARSER_ERROR_HEADER_MESSAGE = (reason: string): string =>
  `Failed to parse grammar: ${reason}`;

export class GrammarParseError extends Error {
  constructor(
    public grammar: string,
    public pos: number,
    public reason: string,
  ) {
    super(
      [
        GRAMMAR_PARSER_ERROR_HEADER_MESSAGE(reason),
        '',
        ...buildErrorPosition(grammar, pos),
      ].join('\n'),
    );
    Object.setPrototypeOf(this, GrammarParseError.prototype);
    this.name = 'GrammarParseError';
  }
}

File: packages/gbnf-experiment/data/20260909T035041Z_2d7fbef6/ported_implementation/gbnf/utils/errors/grammar_parse_error.py
from __future__ import annotations

from ..js import join
from .build_error_position import build_error_position


def GRAMMAR_PARSER_ERROR_HEADER_MESSAGE(reason: str) -> str:
    return f"Failed to parse grammar: {reason}"


class GrammarParseError(Exception):
    def __init__(self, grammar: str, pos: int, reason: str):
        super().__init__(
            join(
                [
                    GRAMMAR_PARSER_ERROR_HEADER_MESSAGE(reason),
                    "",
                    *build_error_position(grammar, pos),
                ],
                "\n",
            )
        )
        self.name = "GrammarParseError"
        self.grammar = grammar
        self.reason = reason
        self.pos = pos

    @property
    def message(self) -> str:
        return str(self)
